layer.go

```diff
@@ -68,9 +67,46 @@ func (p Profile) ProcessAddf(processType string, name string, format string, a .
 	p.Addf(filepath.Join(processType, name), format, a...)
 }
 
-// Contribute represents a layer managed by the buildpack.
-type Layer struct {
+// BOMFormat indicates the format of the SBOM entry
+type BOMFormat int
+
+const (
+	CycloneDXJSON BOMFormat = iota
+	SPDXJSON
+	SyftJSON
+)
+
+func (b BOMFormat) String() string {
+	return []string{"cdx.json", "spdx.json", "syft.json"}[b]
+}
+
+// BOMLocation indicates the location of the SBOM entry
+type BOMLocation int
+
+const (
+	BuildBOM BOMLocation = iota
+	LaunchBOM
+	LayerBOM
+)
+
+// String provides a textual representation
+func (b BOMLocation) String() string {
```

Comment:
When would this be useful? Also is the `LayerBOM` mapping to `<layer>` potentially confusing?

Reply:
I had originally added this with the intent that it would help for logging and error messages, i.e. `%s` would resolve to the textual representation here.

I didn't end up using this in the libpak implementation, so if you're opposed to having it, I'm OK with removing it. 

With that in mind, I opted to have `<layer>` because you can't know the actual value without having access to the layer to which the location applies (since it would be the layer name). This seemed good enough for logging/error messages, but if you feel we should just drop it, I'm OK with that.

Reply:
My latest commit removes this. It seems to be of limited value and potential for confusion, so it's out. We can add something later if there's a demonstrated usecase.